e2e_playwright/st_segmented_control.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import time
+
+import streamlit as st
+
+with st.sidebar:
+    st.markdown(
+        """
+        - [Multi Select - Segmented Control](#multi-select-segmented-control)
+        - [Single Select - Segmented Control](#single-select-segmented-control)
+        - [Icon-only button group - Segmented Control](#icon-only-button-group-segmented-control)
+        - [on_change callback - Segmented Control](#on-change-callback-segmented-control)
+        - [Disabled - Segmented Control](#disabled-segmented-control)
+        - [Segmented Control in form](#segmented-control-in-form)
+        - [Segmented Control in fragment](#segmented-control-in-fragment)
+        - [Unmounted - Segmented Control](#unmounted-segmented-control)
+        """
+    )
+
+st.header("Multi Select - Segmented Control", anchor="multi-select-segmented-control")
+with st.echo(code_location="below"):
```

Comment:
nit: I think it's better not to use `st.echo` as part of our e2e tests. `st.echo` adds a bit of additional complexity and if I remember correctly there are some unexpected side effects if something is wrapped in echo.

Reply:
(This was initially copied from the demo file) I agree that it's better to remove here esp. there are side effects. 
On the side note, as we run all e2e_playwright tests in each CI, the performance of the widget/app should be something to keep in mind right?